src/systemcmds/tests/test_IntrusiveQueue.cpp

```diff
@@ -211,4 +213,59 @@ bool IntrusiveQueueTest::test_push_duplicate()
 	return true;
 }
 
+bool IntrusiveQueueTest::test_remove()
+{
+	IntrusiveQueue<testContainer *> q1;
+
+	// size should be 0 initially
+	ut_compare("size initially 0", q1.size(), 0);
+	ut_assert_true(q1.empty());
+
+	// insert 100
+	for (int i = 0; i < 100; i++) {
+		testContainer *t = new testContainer();
```

Comment:
These are being leaked here, right?

Reply:
Good catch. Fixed in the next commit.
One thing to note from this is you break the iterator if you delete it within the range based for loop.